Fix mypy-ribasim-qgis and add to CI

This would need this commit:
[Add mypy to python_lint github actions](https://github.com/Deltares/Ribasim/pull/1065/commits/8ab4a9076303f1d941adbde158c936264cf380fc)

And the fix to these issues that it finds:
```
✨ Pixi task (default): mypy ribasim_qgis
Warning: --strict-concatenate is deprecated; use --extra-checks instead
ribasim_qgis/tests/test_load_plugin.py:[5](https://github.com/Deltares/Ribasim/actions/runs/7757356113/job/21156633632#step:8:6): error: Class cannot subclass "TestCase" (has type "Any")  [misc]
ribasim_qgis/tests/test_load_plugin.py:12: error: Class cannot subclass "TestCase" (has type "Any")  [misc]
ribasim_qgis/ribasim_qgis.py:[10](https://github.com/Deltares/Ribasim/actions/runs/7757356113/job/21156633632#step:8:11): error: Class cannot subclass "QgsDockWidget" (has type "Any")  [misc]
ribasim_qgis/tests/core/test_model.py:11: error: Class cannot subclass "TestCase" (has type "Any")  [misc]
Found 4 errors in 3 files (checked 22 source files)
```

--- a/.github/workflows/python_lint.yml
+++ b/.github/workflows/python_lint.yml
@@ -30,3 +30,6 @@ jobs:
       - name: Run mypy on python/ribasim_api
         run: |
           pixi run mypy-ribasim-api
+      - name: Run mypy on ribasim_qgis
+        run: |
+          pixi run mypy-ribasim-qgis

--- a/mypy.ini
+++ b/mypy.ini
@@ -10,7 +10,11 @@ strict_concatenate = True
 disallow_subclassing_any = True
 disallow_untyped_decorators = True
 disallow_any_generics = True
-mypy_path=.pixi/env/Library/python:.pixi/env/share/qgis/python
+mypy_path=.pixi/envs/default/Library/python
+
+# Ignore errors for imported packages.
+[mypy-console.*]
+ignore_errors = True
 
 [mypy-qgis.*]
 ignore_errors = True

--- a/mypy.ini
+++ b/mypy.ini
@@ -10,7 +10,7 @@ strict_concatenate = True
 disallow_subclassing_any = True
 disallow_untyped_decorators = True
 disallow_any_generics = True
-mypy_path=.pixi/envs/default/Library/python
+mypy_path=.pixi/envs/default/Library/python:.pixi/envs/default/share/qgis/python
 
 # Ignore errors for imported packages.
 [mypy-console.*]

